src/io.c

```diff
@@ -417,3 +417,41 @@ niova_io_iov_restore(struct iovec *iovs, size_t niovs, size_t save_idx,
 
     return 0;
 }
+
+/**
+ * iovec iterator - calls callback once per iovec.
+ *
+ * @iovs:      Array of iovecs containing the data
+ * @niovs:     Number of iovecs in the array
+ * @num_bytes: Number of bytes to iterate over
+ * @cb:        Callback function invoked for each iovec segment
+ * @user_ctx:  Context pointer passed through to callback
+ *
+ */
+ssize_t
+niova_io_iterate_iovs(const struct iovec *iovs, size_t niovs,
+                      size_t num_bytes, niova_iov_cb_t cb, void *user_ctx)
```

Comment:
We need a unit test for this.  Can you think of something that would allow us to verify that iteration is occurring as expected?  The test should loop using randomized parameters at each iteration.  A few thousand iteration should be fine..